### PR TITLE
expression: implement vectorized evaluation for `builtinRealIsTrueSig`

### PR DESCRIPTION
**Pull request by liuhao2050**
Implement vectorized evaluation for builtinRealIsTrueSig.
#12103

### What problem does this PR solve? 
```
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinOpFunc/builtinRealIsTrueSig-VecBuiltinFunc-4            335839              3479 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinOpFunc/builtinRealIsTrueSig-NonVecBuiltinFunc-4          47275             26093 ns/op               0 B/op          0 allocs/op
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_op_vec.go [hunk 1]
```diff
@@ -366,11 +366,31 @@ func (b *builtinRightShiftSig) vecEvalInt(input *chunk.Chunk, result *chunk.Colu
...
 
 func (b *builtinRealIsTrueSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinRealIsTrueSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	numRows := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETInt, numRows)
```
Comment (Reminiscent) [inline expression/builtin_op_vec.go]: Here should be ```buf, err := b.bufAllocator.get(types.ETReal, numRows)```

Comment (liuhao2050) [reply expression/builtin_op_vec.go]: My mistake, it's now fixed.

**Comment by XuHuaiyu** [comment]
@liuhao2050 Please resolve the conflicts.

**Pull request merged**